packages/ui/feature-builder-right-sidebar/src/lib/input-forms/piece-input-form/piece-input-form.component.ts

```diff
@@ -0,0 +1,380 @@
+            (curr, prev) => curr?.stepName === prev?.stepName,
+          ),
+          tap((step) => {
+            this.triggersOrActionsControl.setValue(
+              step.triggerOrActionname || '',
+              { emitEvent: false },
+            );
+          }),
+          switchMap((step) => {
+            return this.triggersOrActionsControl.valueChanges.pipe(
+              startWith(step.triggerOrActionname || ''),
+            );
+          }),
+        ),
+      triggersOrActions: this.getTriggersOrActions(),
+      pieceMetaData: this.getPieceMetaData(),
+    };
+    return combineLatest(deps$).pipe(
+      map((res) => {
+        const triggerOrAction = res.triggersOrActions.find(
+          (v) => v.name === res.selectedTriggerOrActionName,
+        );
+        if (triggerOrAction) {
+          return addPieceAuthenticationPropertyToTriggerOrActionProperties(
+            triggerOrAction,
+            res,
+          );
+        }
+        return undefined;
+      }),
+    );
+
+    function addPieceAuthenticationPropertyToTriggerOrActionProperties(
+      triggerOrAction: ActionBase | TriggerBase,
+      res: {
+        selectedTriggerOrActionName: string;
+        triggersOrActions: ActionBase[] | TriggerBase[];
+        pieceMetaData: PieceMetadataModel | undefined;
+      },
+    ) {
+      const selected = {
+        ...triggerOrAction,
+        props: {
+          ...spreadIfDefined(
+            AUTHENTICATION_PROPERTY_NAME,
+            res.pieceMetaData?.auth,
+          ),
+          ...triggerOrAction.props,
+        },
+      };
+      return selected;
+    }
+  }
+  getPieceMetaData(): Observable<PieceMetadataModel | undefined> {
+    const currentStep$ = this.store.select(BuilderSelectors.selectCurrentStep);
+    return currentStep$.pipe(
+      switchMap((step) => {
+        if (!step) {
+          return of(undefined);
+        }
+        if (step.type === ActionType.PIECE || step.type === TriggerType.PIECE) {
+          return this.pieceMetaDataService.getPieceMetadata(
+            step.settings.pieceName,
+            step.settings.pieceVersion,
+          );
+        }
+        return of(undefined);
+      }),
+    );
+  }
+
+  getAllConnectionsForPiece() {
+    const currentStep$ = this.store.select(BuilderSelectors.selectCurrentStep);
+    return currentStep$.pipe(
+      switchMap((step) => {
+        if (
+          !step ||
+          (step.type !== ActionType.PIECE && step.type !== TriggerType.PIECE)
+        )
+          return of([]);
+        return this.store.select(
+          appConnectionsSelectors.selectAllConnectionsForPiece(
+            step.settings.pieceName,
+          ),
+        );
+      }),
+    );
+  }
+
+  piecePropertiesFormValueChanged(
+    result: {
+      input: Record<string, unknown>;
+      customizedInputs: Record<string, boolean | Record<string, boolean>>;
+      valid: boolean;
+    },
+    step: Step,
+  ) {
+    debugger;
+    if (step.type === TriggerType.PIECE) {
```

Comment:
Switch better there 
----
(Can be done later)
Another suggestion i would create action in store to update input 

Please note that would break the mirroring thing in the state

Reply:
mirroring thing ?